Library/Formula/ghc.rb

```diff
@@ -15,10 +15,10 @@ class Ghc < Formula
   deprecated_option "tests" => "with-tests"
 
   resource "gmp" do
-    url "http://ftpmirror.gnu.org/gmp/gmp-6.0.0a.tar.bz2"
-    mirror "ftp://ftp.gmplib.org/pub/gmp/gmp-6.0.0a.tar.bz2"
-    mirror "ftp://ftp.gnu.org/gnu/gmp/gmp-6.0.0a.tar.bz2"
-    sha256 "7f8e9a804b9c6d07164cf754207be838ece1219425d64e28cfa3e70d5c759aaf"
+    url "http://ftpmirror.gnu.org/gmp/gmp-6.0.0a.tar.xz"
+    mirror "https://ftp.gnu.org/gnu/gmp/gmp-6.0.0a.tar.xz"
```

Comment:
As a side addition: `ftp` is a nightmare to get through tight firewalls. It's the reason I've been religiously sticking `http` or `https` mirrors onto every GNU formulae I deal with :smile_cat:.